src/components/media/RemoveImage.tsx

```diff
@@ -0,0 +1,41 @@
+import { actions } from '../../js/stores'
+import { removePhoto } from '../../js/userApi/media'
+import { MediaType } from '../../js/types'
+import { useSession } from 'next-auth/react'
+import useReturnToProfile from '../../js/hooks/useReturnToProfile'
+
+interface RemoveImageProps {
+  imageInfo: MediaType
+  tagCount: Number
+}
+
+export default function RemoveImage ({ imageInfo, tagCount }: RemoveImageProps): JSX.Element | null {
+  const { data } = useSession()
+  const { toMyProfile } = useReturnToProfile()
+
+  const onRemove = async (e): Promise<void> => {
+    if (data?.user?.metadata == null) {
+      console.log('## Error: user metadata not found')
+      return
+    }
+    const { nick } = data.user.metadata
+
+    const filename: string = imageInfo.filename
+    e.preventDefault()
+    if (window.confirm('Are you sure?')) {
+      await actions.media.removeImage(imageInfo.mediaId, nick)
+      await removePhoto(filename)
+      await toMyProfile()
```

Comment:
In this case the full screen state is still active in the parent gallery component so it's loading a different photo (the current index has changed after a removal). I'd remove this and implement a callback as suggested to notify parent component.